src/components/Modal/_helpers/dialogOnClickHandler.js

```diff
@@ -0,0 +1,34 @@
+// Disable coverage for the following function
+/* istanbul ignore next line */
+export const dialogOnClickHandler = (
+  e,
+  closeButtonRef,
+  dialogRef,
+  allowCloseOnBackdropClick,
+) => {
+  // If it is not allowed to close modal on backdrop click, do nothing.
+  if (!allowCloseOnBackdropClick) {
+    return;
+  }
+
+  // Detection of the click on the backdrop is based on the following conditions:
+  // 1. The click target is the dialog itself. This prevents detection of clicks on the dialog's children.
+  // 2. The click is outside the dialog's boundaries.
+  const dialogRect = dialogRef.current.getBoundingClientRect();
+  const isClickedOnBackdrop = dialogRef.current === e.target && (
+    e.clientX < dialogRect.left
+    || e.clientX > dialogRect.right
+    || e.clientY < dialogRect.top
+    || e.clientY > dialogRect.bottom
+  );
```

Comment:
Do we still need this with native dialog? The `<dialog>` triggers the `close` event so I was wondering if we could listen on it?

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#handling_the_return_value_from_the_dialog

Reply:
If it wasn't needed, it wouldn't be here. Neither close nor cancel event is called on backdrop click. If you want to close dialog by clicking on backdrop, you have to listen to click event and detect user clicked outside of dialog element or use different method of detection. However, this is most common approach.

It is quite hard to decide what to comment. This is default behavior/knowledge and developer is supposed to know this if he/she want to dig into it. Now, each line of each event handler is commented. But I would add docblock for each function to make it crystal clear.

Reply:
I added docblock for those handlers. It still require knowledge, but it is easier to understand now. Most problematic part is Escape click that can end-up in 3 handles, but it is described in native documanation.